src/Diagnostics.RuntimeHost/Controllers/ProcessHealthController.cs

```diff
@@ -13,18 +15,44 @@ public class ProcessHealthController : Controller
     {
         private ISourceWatcherService _sourceWatcherService;
         private ICompilerHostClient _compilerHostClient;
+        private IConfiguration _configuration;
 
-        public ProcessHealthController(ISourceWatcherService sourceWatcherService, ICompilerHostClient compilerHostClient)
+        public ProcessHealthController(ISourceWatcherService sourceWatcherService, ICompilerHostClient compilerHostClient, IConfiguration Configuration)
         {
             // These dependencies are injected for the services to start.
             _sourceWatcherService = sourceWatcherService;
             _compilerHostClient = compilerHostClient;
+            _configuration = Configuration;
+        }
+
+        private static bool PingExternal(string pingUrl)
+        {
+            try
+            {
+                using (var client = new WebClient())
+                using (client.OpenRead(pingUrl))
+                    return true;
+            }
+            catch
+            {
+                return false;
+            }
         }
 
         [HttpGet(UriElements.HealthPing)]
         public async Task<IActionResult> HealthPing()
         {
             await _sourceWatcherService.Watcher.WaitForFirstCompletion();
+
+            bool isCheckEnabled = Convert.ToBoolean(_configuration["OutboundConnectivity:IsCheckEnabled"]);
```

Comment:
Again, this can be moved to the HealthCheckService and you can read this setting once in ctor and maintain it in a local variable.